packages/core-data/src/actions.js

```diff
@@ -139,6 +139,70 @@ export function receiveEmbedPreview( url, preview ) {
 	};
 }
 
+/**
+ * Action triggered to delete an entity record.
+ *
+ * @param {string} kind      Kind of the deleted entity.
+ * @param {string} name      Name of the deleted entity.
+ * @param {Object} recordId  Record to be deleted.
+ */
+export function* deleteEntityRecord( kind, name, recordId ) {
+	const entities = yield getKindEntities( kind );
+	const entity = find( entities, { kind, name } );
+	if ( ! entity ) {
+		return;
+	}
+
+	yield {
+		type: 'DELETE_ENTITY_RECORD_START',
+		kind,
+		name,
+		recordId,
+	};
+
+	yield removeItems( kind, name, Number( recordId ) );
+
+	let error;
+
+	try {
+		let path = `${ entity.baseURL }/${ recordId }`;
+
+		if ( entity.forceDelete ) {
+			path += '?force=true';
+		}
+
+		yield apiFetch( {
+			path,
+			method: 'DELETE',
+		} );
+	} catch ( _error ) {
+		error = _error;
+
+		const persistedRecord = yield select(
+			'getEntityRecord',
+			kind,
+			name,
+			recordId
+		);
+
+		yield receiveEntityRecords(
+			kind,
+			name,
+			persistedRecord,
+			undefined,
+			false
+		);
+	}
```

Comment:
If it errors, we also need to restore the queries updated by `removeItems`.

I think we also should restore the edited record, not the persisted one.

Reply:
> I think we also should restore the edited record, not the persisted one.

This basically means:
```
const persistedRecord = yield select(
			'getEditedEntityRecord',
			kind,
			name,
			recordId
		);
```
Right?